extensions/elytron-security/deployment/pom.xml

```diff
@@ -31,6 +31,10 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-elytron-security</artifactId>
         </dependency>
+        <dependency>
+          <groupId>io.quarkus</groupId>
+          <artifactId>quarkus-agroal-deployment</artifactId>
+        </dependency>
```

Comment:
Why is the agroal dependency needed ? Should the user application pull the `quarkus-agroal` instead ?

Reply:
🤔 I  used the flyway extension as inspiration source.... At this point, I don't know yet if elytron extension will need to manipulate the Agroal Datasource object (or if everything will be done at runtime). 

You're right I should don't couple the 2 extensions).

Reply:
I hope so, at runtime :-), my understanding the other DB related extensions depend on Agroal

Reply:
At the moment, I flagged optional.

IMHO, in real life there are more application using DBs as Identity Store rather than files. 
From a developer point of view, he/she uses a DB for authentication,  I'm almost sure,  It will  use hibernate/ flyway somewhere in the application. These dependencies actually includes Agroal but to make authentication work the dev will have to re-declare explicitly Agroal.

I understand the reason but the dependency organization looks a bit weird to me. 

For me the 2 options are:
- Include Agroal by default by the the security extension
- Have a dedicated extension for JDBC realm, which makes sens since I had to import the dedicated `wildfly-elytron-realm-jdbc`.